avm/res/document-db/database-account/main.bicep

```diff
@@ -367,6 +403,49 @@ module databaseAccount_privateEndpoints 'br/public:avm/res/network/private-endpo
   }
 }]
 
+module keyVault 'modules/secrets-key-vault.bicep' = if (secretsKeyVaultReference != null) {
+  name: '${uniqueString(deployment().name, location)}-secrets-kv'
+  scope: resourceGroup(secretsKeyVaultReference.?resourceGroupName ?? resourceGroup().name)
+  params: {
+    keyVaultName: secretsKeyVaultReference!.keyVaultName
+
+    keySecrets: [
+      {
+        secretName: secretsKeyVaultReference.?primaryWriteKeySecretName ?? 'Primary-Write-Key'
+        secretValue: databaseAccount.listKeys().primaryMasterKey
```

Comment:
Idea: Instead of creating this list here, you could also do it inside the `secrets-key-vault.bicep` file by passing the databaseAccount resourceId and implementing an `existing` reference.
The advantage: You don't have to pass any secret in between scopes, nor do you have to implement a user defined type.
The disadvantage: You'd be restricted to only those secrets and would not be able to pass additional ones in.

Reply:
For this one, I started to do it but I really don't like the end result and I wouldnt like to implement it. I still will have to define the UDT in the inner module as the user needs to be able to change the secret name so I need to pass those parameters down to it. I can always not implement the UDT in the inner module and pass everything as normal parameters, but it will be uglier than what we have now. I dont mind passing secrets between scopes. I already added the secure descriptor to the UDT. When we have exportable types the conversation will be different.

Reply:
Hey @bryansan-msft, thanks for elaborating 💪 Makes sense